100-navigation

ceciliaccwei opened the pull request:
Summary:

- Route to homepage after login.
- Save username and userid after login
- Set up router to navigate between UIs
- Updated props from some UIs
- Load username in edit profile modal; Show username in navigation bar and profile page
- PrivateRoute (Fixed using #124 ) - Logic for this component is if the user has logged in, we redirect to homepage instead of asking him to log in again. _Will create another issue for this after M1._ Right now, due to the response header in the 401 response, an auth window will always pop up to prompt for the user's credentials. 
I noticed sometimes it takes some time for the end point to get updated with the current user. But the redirection works now without that ugly auth popup window(yaaaaaaaas!!!)
- Get relationship between two people to render the corresponding profile page header
- Fixed Issue where the app did not update when viewing self's profile from other person's profile
- Removed extra paddings

maharshmellow commented on src/components/profile/ProfilePage.jsx:
```diff
@@ -1,35 +1,97 @@
+              {this.renderHeader()}
+            </div>
+            <PostView
             // TODO: change this to the current user's full id
```
we can get rid of this comment now 😄

ceciliaccwei replied:
done!

maharshmellow commented on src/components/profile/ProfilePage.jsx:
```diff
@@ -1,35 +1,97 @@
+    if (!isSelf) {
+      friendsService.checkFriendStatus(currentUserID, userID).then((response) => {
+        if (response) {
+          friendsService.checkFriendStatus(userID, currentUserID).then((response2) => {
```
based on my conversation with Roy, `checkFriendStatus(A, B)` is the same as `checkFriendStatus(B, A)` so I think we can get rid of this second check here

ceciliaccwei replied:
In my testing, 
Friends- fromUser:3, toUser:1
http://127.0.0.1:8000/author/https://cmput404-socialdistribution.herokuapp.com/author/3/friends/https://cmput404-socialdistribution.herokuapp.com/author/1 -> true;
http://127.0.0.1:8000/author/https://cmput404-socialdistribution.herokuapp.com/author/1/friends/https://cmput404-socialdistribution.herokuapp.com/author/3 -> false;
Based on the behaviour, I think that endpoint is checking if A is following B.

maharshmellow replied:
Interesting.. I thought follows didn't even show up in the friends table and were only in the friend requests table. @roychowd @sandysaji can you look at this?

roychowd replied:
hey @ceciliaccwei  @maharshmellow  @sandysaji. Please look at https://github.com/CMPUT404W20/CMPUT404-project-socialdistribution/pull/128 

There was a bug that only checked for one way friendship, not two-way.

roychowd replied:
@ceciliaccwei @maharshmellow @sandysaji. I fixed this issue. It was not checking for two-way friendship. Please see https://github.com/CMPUT404W20/CMPUT404-project-socialdistribution/pull/129

maharshmellow commented on src/services/AuthenticationService.jsx:
```diff
@@ -15,3 +15,7 @@ export const registerUser = (username, password) => {
 };
+
+export const getCurrentUser = () => {
+  return axios.get("/author/current");
```
should we check for a status 200 here (like how we do in the PostService)? We didn't add a check for the other 2 functions in this file but we can do that later.

ceciliaccwei replied:
It is checked in PrivateRoute when I use it. 
[ auth.getCurrentUser()
      .then(((response) => {
        if (response.status === 200) {...](url)

maharshmellow commented on src/styles/NavigationBar.scss:
```diff
@@ -1,104 +1,87 @@
-      }
+    background-color: white;
+    box-shadow: 0px 0px 10px rgba(121, 121, 121, 0.25);
+    ;
```
🙈

ceciliaccwei replied:
Opps removed extra semi-colon

ceciliaccwei commented:
Updates:

- updated friends check based on #128 
- changed profile header logic based on api update;
- saved userid into local storage after fetching from current user endpoint
- updated profile header image to match theme
- comments cleaned up
